sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -692,6 +693,31 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         tasks.RemoveAll(t => t.IsCompleted);
                     }
                 }
+
+                // If the main loop is aborting due to the connection being canceled, then
+                // force the processor to stop.
+                if (!cancellationToken.IsCancellationRequested && Connection.IsClosed)
+                {
+                    Logger.ProcessorClientClosedException(Identifier);
+
+                    // Because this is a highly unusual situation
+                    // and goes against the goal of resiliency for the processor, surface
+                    // a fatal exception with an explicit message detailing that processing
+                    // will be stopped.
+                    var fatalException = new ObjectDisposedException(
+                        nameof(ServiceBusConnection),
+                        Resources.DisposedConnectionMessageProcessorMustStop);
+
+                    await InvokeErrorHandler(fatalException, ServiceBusErrorSource.Receive).ConfigureAwait(false);
```

Comment:
Any reason to not include the cancellation token? I don't think it would have a functional impact to include it as we know it is not cancelled.

Reply:
Good catch.   The ordering was a bit different once upon a time;  I think we can safely pass it at this point.

Reply:
Now that this is only called from one place, we might just inline the body of the method here.